[DOC-12901]: 7.6.4 release notes incomplete for Search service

RayOffiah opened the pull request:
- [x] Updated 7.6.4 release notes for Search service
- [x] Implemented pre-filtering for vector search queries
- [x] Added documentation links for new feature

RayOffiah committed:
[DOC-12901]: 7.6.4 release notes incomplete for Search service

Add pre-filtering support for vector search queries

The Search Service now supports pre-filtering in vector search, improving precision and performance. Documentation links for this new feature have been included for further details.

ElliotFrancisHunter reviewed (approved):
LGTM 👍🏼

abhinavdangeti commented on modules/introduction/partials/new_features-76_4.adoc:
```diff
@@ -30,6 +30,16 @@ curl --get -u <username:password> \
++
+--
+** Enhanced precision and relevance in the search results.
+** Performance optimization.
```
Ok so it's not a performance optimization - because we'd be running the search request in 2 phases for this.

What pre-filtering gives is the ability to be able to run vector search over a subset of the indexed data. The "filter" is responsible for qualifying this subset of data.

RayOffiah replied:
Does that mean we should remove `performance optimization` from the main pre-filtering page?

https://docs.couchbase.com/server/current/vector-search/pre-filtering-vector-search.html#about-pre-filtering

abhinavdangeti replied:
Yes please, would you update both pages.

RayOffiah replied:
Okay, so what advantage does this give the user?

Is it about enhancing the precision of the search?

abhinavdangeti replied:
No, I just put the benefit in my comment above^.

RayOffiah replied:
No worries; I'll figure it out.

RayOffiah committed:
Enhance Search Service with vector query pre-filtering

Pre-filtering enables executing vector searches on a subset of the index, optimizing computational resources and improving efficiency. Updated documentation to reflect this enhancement and its benefits.

abhinavdangeti commented on modules/introduction/partials/new_features-76_4.adoc:
```diff
@@ -30,6 +30,12 @@ curl --get -u <username:password> \
 
+* The Search Service now supports pre-filtering on vector search queries. 
+Pre-filtering allows you to execute vector searches over a subset of the vector index, 
+which can reduce the computational resource required to complete the query.
```
It does not reduce the compute required for the query. Let's simply replace this with ..

```
* The Search Service now supports pre-filtering on vector search queries. 
Pre-filtering allows you to execute vector searches over a subset of the vector index, 
via the means of a filter request that qualifies the subset.
```

RayOffiah committed:
Enhance Search Service with vector query pre-filtering

Pre-filtering enables executing vector searches on a subset of the index, optimizing computational resources and improving efficiency. Updated documentation to reflect this enhancement and its benefits.

sarahlwelton commented on modules/introduction/partials/new_features-76_4.adoc:
```diff
@@ -30,6 +30,11 @@ curl --get -u <username:password> \
 [#new-features-764-search-service]
 === Search Service
 
+* The Search Service now supports pre-filtering on vector search queries. 
```
```suggestion
* The Search Service now supports pre-filtering on Vector Search queries. 
```

sarahlwelton commented on modules/introduction/partials/new_features-76_4.adoc:
```diff
@@ -30,6 +30,11 @@ curl --get -u <username:password> \
 === Search Service
 
+* The Search Service now supports pre-filtering on vector search queries. 
+  Pre-filtering allows you to execute vector searches over a subset of the vector index, 
```
```suggestion
  Use pre-filtering to execute a vector search over a subset of your Vector Search index, 
```

sarahlwelton commented on modules/introduction/partials/new_features-76_4.adoc:
```diff
@@ -30,6 +30,11 @@ curl --get -u <username:password> \
 
+* The Search Service now supports pre-filtering on vector search queries. 
+  Pre-filtering allows you to execute vector searches over a subset of the vector index, 
+  via the means of a filter request that qualifies the subset. +
```
```suggestion
  through a defined filter request. +
```

sarahlwelton reviewed (changes requested):
Just some language tweaks. Thanks for taking this up!

RayOffiah committed:
Enhance Search Service with vector query pre-filtering

Pre-filtering enables executing vector searches on a subset of the index, optimizing computational resources and improving efficiency. Updated documentation to reflect this enhancement and its benefits.
